### PR TITLE
[MAINT] use pathlib.Path more systematically in nilearn.datasets

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4327 

Changes proposed in this pull request:

- use pathlib.Path more systematically in nilearn.datasets
- still need to do fetch_files: in another PR

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
Great, thx !

File: nilearn/datasets/func.py [hunk 1]
```diff
@@ -785,9 +786,8 @@ def fetch_localizer_contrasts(
                 name_aux = str.replace(
                     str.join("_", [data_type, contrast]), " ", "_"
                 )
-                file_path = Path(
-                    "brainomics_data", subject_id, f"{name_aux}.nii.gz"
-                )
+                file_path = f"brainomics_data/{subject_id}/{name_aux}.nii.gz"
```
Comment (michellewang) [inline nilearn/datasets/func.py:789]: I'm sure you had a good reason but, for my own understanding, for revert to hardcoded `"/"`?

**Review by michellewang — APPROVED** [review]
Minor comment (question) which is really for my own understanding, otherwise looks good!

File: nilearn/datasets/func.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/datasets/func.py:789]: hmm... I thing I was getting some tests failures and tried to fix it but forgot to revert those.
let me fix this

**Pull request merged**